packages/static-hosting/lib/static-hosting.ts

```diff
@@ -425,23 +435,39 @@ export class StaticHosting extends Construct {
     });
     let backendOrigin = undefined;
 
+    const originRequestHeaderBehaviorAllowList = [
+      "x-forwarded-host", // Consumed by OriginRequest Lambda@Edge for Feature Environment functionality.
+      "x-request-prerender", // Consumed by OriginRequest Lambda@Edge to determine if this request needs to be send to Prerender service rather than other origins.
+      "x-prerender-host", // Consumed by OriginRequest Lambda@Edge, only when x-request-prerender header is set. Prerender service will send request to this host.
+      "x-prerender", // Consumed, if configured, by origin's custom features, such as GeoRedirection, the behave of which should depend on whether the request is from an end user.
+      "x-prerender-user-agent", // Consumed by Prerender service for logging original user agent rather than CloudFront's
+    ];
```

Comment:
Instead of having the for each and pushing you can use the spread operator:
```
const originRequestHeaderBehaviorAllowList = [
  "x-forwarded-host",
  "everything else....",
  ...additionalDefaultOriginRequestHeaders,
]
```
This does the same thing as the for loop but is cleaner. (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)

You may need a line above that does this though:
```
const additionalDefaultOriginRequestHeaders = props.additionalDefaultOriginRequestHeaders || []
```
Just to make sure the value is always defined.